src/gui/torrentcontentmodel.cpp

```diff
@@ -65,6 +65,39 @@
 
 namespace
 {
+    class FolderAdaptor
+    {
+    public:
+        using FolderAdaptorPtr = std::shared_ptr<FolderAdaptor>;
+
+        explicit FolderAdaptor(TorrentContentModelFolder *folder)
+            : m_folder {folder}
+        {
+        }
+
+        FolderAdaptorPtr findChildFolderOrInsert(const QString &folderName)
+        {
+            FolderAdaptorPtr folder = m_folders.value(folderName);
+            if (folder)
+                return folder;
+            folder = std::make_shared<FolderAdaptor>(new TorrentContentModelFolder(folderName, m_folder));
+            m_folder->appendChild(folder->m_folder);
+            m_folders.insert(folderName, folder);
+            return folder;
+        }
+
+        TorrentContentModelFile *insertFile(const QString &fileName, const qulonglong fileSize, const int fileIndex)
+        {
+            auto file = new TorrentContentModelFile(fileName, fileSize, m_folder, fileIndex);
+            m_folder->appendChild(file);
+            return file;
+        }
+
+    private:
+        TorrentContentModelFolder *m_folder;
+        QHash<QString, FolderAdaptorPtr> m_folders;
+    };
```

Comment:
```suggestion
    class FolderAdaptor
    {
    public:
        using FolderAdaptorPtr = std::shared_ptr<FolderAdaptor>;

        explicit FolderAdaptor(TorrentContentModelFolder *folder, QVector<TorrentContentModelFile *> &fileIndex)
            : m_folder {folder}
            , m_files {fileIndex}
        {
        }

        FolderAdaptorPtr findChildFolderOrInsert(const QString &folderName)
        {
            FolderAdaptorPtr folder = m_folders.value(folderName);
            if (folder)
                return folder;
            folder = std::make_shared<FolderAdaptor>(new TorrentContentModelFolder(folderName, m_folder), m_files);
            m_folder->appendChild(folder->m_folder);
            m_folders.insert(folderName, folder);
            return folder;
        }

        void insertFile(const QString &fileName, const qulonglong fileSize, const int fileIndex)
        {
            auto file = new TorrentContentModelFile(fileName, fileSize, m_folder, fileIndex);
            m_folder->appendChild(file);
            m_files.append(file);
        }

    private:
        TorrentContentModelFolder *m_folder;
        QHash<QString, FolderAdaptorPtr> m_folders;
        QVector<TorrentContentModelFile *> m_files;
    };
```

Reply:
why?

Reply:
To adapt all the related stuff. Then you just do:
`auto rootFolderAdaptor = std::make_shared<FolderAdaptor>(m_rootItem, m_filesIndex);` and then `currentParent->insertFile(Utils::Fs::fileName(info.filePath(i)), info.fileSize(i), i);`

Reply:
I think it is fine as it is, I don't know if it is worth it to have a reference and add more responsibility to a class for saving one line.

Reply:
OK, I don't insist.

P.S. the number of lines is not the best measure generally.